text/0000-ppv.md

```diff
@@ -0,0 +1,1721 @@
+  combination with the stabilization of the `std::arch` intrinsics for those
+  architectures.
+
+### Layout of vector types
+
+The portable packed SIMD vector types introduced in this RFC are layout
+compatible with the architecture-specific vector types. That is:
+
+```rust
+union A {
+   port: f32x4,
+   arch: __m128,
+}
+let x: __m128 = _mm_setr_ps (0.0, 1.0, 2.0, 3.0);
+let y: f32x4 = A { arch: x }.port;
+assert_eq!(y.extract(0), 0.0);  // OK
+assert_eq!(y.extract(1), 1.0);  // OK
+assert_eq!(y.extract(2), 2.0);  // OK
+assert_eq!(y.extract(3), 3.0);  // OK
+```
+
+The portable packed SIMD vector types are also layout compatible with arrays of
+equal element type and whose length equals the number of vector lanes. That is:
+
+```rust
+union A {
+   port: f32x4,
+   arr: [f32; 4],
+}
+let x: [f32; 4] = [0.0, 1.0, 2.0, 3.0];
+let y: f32x4 = A { arr: x }.port;
+assert_eq!(y.extract(0), 0.0);  // OK
+assert_eq!(y.extract(1), 1.0);  // OK
+assert_eq!(y.extract(2), 2.0);  // OK
+assert_eq!(y.extract(3), 3.0);  // OK
+```
+
+This transitively makes both portable packed and architecture specific SIMD
+vector types layout compatible with all other types that are also layout
+compatible with these array types.
+
+## API of portable packed SIMD vector types
+
+### Traits overview
+
+All vector types implement the following traits:
+
+* `Copy`
+* `Clone`
+* `Default`: zero-initializes the vector.
+* `Debug`: formats the vector as `({}, {}, ...)`.
+* `PartialEq<Self>`: performs a lane-wise comparison between two vectors and
+  returns `true` if all lanes compare `true`. It is equivalent to
+  `a.eq(b).all()`.
+* `PartialOrd<Self>`: compares two vectors lexicographically.
+* `From/Into` lossless casts between vectors with the same number of lanes.
+
+All signed integer, unsigned integer, and floating point vector types implement
+the following traits:
+
+* `{Add,Sub,Mul,Div,Rem}<RHS=Self,Output=Self>`,
+  `{Add,Sub,Mul,Div,Rem}Assign<RHS=Self>`: vertical (lane-wise) arithmetic
+  operations.
+
+All signed and unsigned integer vectors and vector masks also implement:
+
+* `Eq`: equivalent to `PartialEq<Self>`
+* `Ord`: equivalent to `PartialOrd<Self>`
+* `Hash`: equivalent to `Hash` for `[element_type; number_of_elements]`.
+* `fmt::LowerHex`/`fmt::UpperHex`: formats the vector as hexadecimal.
+* `fmt::Octal`: formats the vector as an octal number.
+* `fmt::Binary`: formats the vector as binary number.
+* `Not<Output=Self>`: vertical (lane-wise) negation,
+* `Bit{And,Or,Xor}<RHS=Self,Output=Self>`, `Bit{And,Or,Xor}Assign<RHS=Self>`:
+  vertical (lane-wise) bitwise operations.
+
+All signed and unsigned integer vectors also implement:
+
+* `{Shl,Shr}<RHS=Self,Output=Self>`, `{Shl,Shr}Assign<RHS=Self>`: vertical
+  (lane-wise) bit-shift operations.
+
+Note: While IEEE 754-2008 provides total ordering predicates for floating-point
+numbers, Rust does not implement `Eq` and `Ord` for the `f32` and `f64`
```

Comment:
But it does [Implement total_cmp for f32, f64](https://github.com/rust-lang/rust/pull/72568). Maybe I'm misunderstanding something?

Reply:
Also, f32xN and f64xN should implement the bitwise ops.